docs/sections/calculations/errors.md

```diff
@@ -565,346 +222,107 @@ $ verdi calcjob outputcat <pk_number> | less
+Use `verdi process list -a -p1` to verify that the error code is different now.
+You can check again the outputs and the reports with the tools explained in this section and try to fix it yourself before going on to the next.
 
-```{code-block} python
 
-parameters['CONTROL']['calculation'] = 'vc-relax'
+## Restarting calculations
 
-```
+It turns out that your last calculation did not converge because we stopped the self-consistency iteration cycle before it converged (by only setting a max of 3 cycles).
+In this simple case, you could just re-run the calculation from scratch with a sufficient number of iterations, but for expensive calculations (including a structural relaxation or molecular dynamics), you would like instead to restart your calculation from the previous one to save time.
 
-And once again check the `TrajectoryData` output node with link name `output_trajectory`.
+For this purpose, `CalcJobNode` provides the `get_builder_restart` method.
+Just like the `get_builder` method of the `Code` or of the `Process` class, this creates an instance of the `ProcessBuilder`, but with all inputs already pre-populated from those of the "parent" node.
 
-The output of calculation jobs can also be obtained via the `verdi shell`.
-For example, note down the PK of the calculation so that you can load it in the `verdi shell` and check the total energy with the commands:
+Let us load the node of the calculation job that we want to restart in a `verdi shell` and create a new builder from it:
 
 ```{code-block} ipython
 
-In [1]: calculation = load_node(<PK>)
+In [10]: failed_calculation = load_node(<PK>)
+    ...: restart_builder = failed_calculation.get_builder_restart()
 
 ```
 
-Then get the energy of the calculation with the command:
+Type `restart_builder` and press Enter to verify that all inputs have already been set to those that were used for the original calculation.
+Let's give the new calculation some more steps in the SCF cycle in order to let it converge:
 
 ```{code-block} ipython
 
-In [2]: calculation.res.energy
-Out[2]: -308.31214469484
+In [11]: parameters = restart_builder.parameters.get_dict()
+    ...: parameters['ELECTRONS']['electron_maxstep'] = 80
 
 ```
 
-You can also type
+The `aiida-quantumespresso` plugin supports restarting a calculation by setting the corresponding `restart_mode` and attaching the remote working directory of the previous calculations as the `parent_folder` input [^f1]:
 
 ```{code-block} ipython
 
-In [3]: calculation.res.
+In [12]: parameters['CONTROL']['restart_mode'] = 'restart'
+    ...: restart_builder.parent_folder = failed_calculation.outputs.remote_folder
+    ...: restart_builder.parameters = Dict(dict=parameters)
 
 ```
 
-and then press `TAB` to see all the available results of the calculation.
+Note that we've created a new `Dict` node for the modifed parameters since the original input is already stored in the database and is therefore immutable.
```

Comment:
Not relevant information in this context.
```suggestion
```

Reply:
If I understand correctly, you mean that immutability is not a concept related to the running or debugging of calculations and should not be introduced / explained in this section section. If so, I completely agree with you.

However, I would argue that that is not exactly what I'm doing here. I think there is no other place in the tutorial where the practical implications of the immutability of nodes are illustrated: here is the only place where the user has to actually modify some content from a node in order to use it for other thing. It is the natural (and only, at least in this tutorial) place where the user would wonder "why can't I just modify this node directly?". This is in my opinion the best and "canonical" case where the immutability of the node impacts how the user works and it is very valuable to point it out.

I could take it out of the main text and put it in an admonition, but I'm also tempted to have it be an "Important" box rather than a simple "Note" box.

Reply:
You could've written this sentence multiple times throughout this exercise, why now? Why is it important to have it stated now, when we've already created multiple different immutable `Dict` objects prior in the same exercise? What makes this point in the exercise special?

But I agree that it would be good if the reader realizes what you write above. So an "important" admonition might be good. But also mention that this was then the case when the parameters was changed previously to showcase the erroneous QE parameters.

Reply:
> You could've written this sentence multiple times throughout this exercise, why now?

I think "framing" is the key difference. You rightfully point towards the previous correction which can _technically_ be understood as the same thing, but it is not _framed_ as such: note that they don't take the info from the node they created previously, we are telling them to create a brand new node with a brand new dictionary.

It is true that they might wonder "why do I need to create a new one instead of correcting the old one?" but I feel this is less "dissonant" since creating a new one from scratch just like they did before, there is no modification over any pre-existing variable, it is just an alternative way of doing the thing. But in this case they are doing a modification on something that was already on a node (which moreover is already plugged into the builder), why not do it directly on the node? Creating a new one does not feel like an alternative, but more like an unnecessary extra step.

Reply:
Very true. It might be nice to make a nod back to the previous section to point this out, but I agree completely on all your points here.

Maybe just make it an admonition then? :)